src/bun.js/node/node_crypto_binding.zig

```diff
@@ -25,3 +26,89 @@ pub fn randomInt(global: *JSC.JSGlobalObject) callconv(.C) JSC.JSValue {
     };
     return JSC.JSFunction.create(global, "randomInt", S.cb, 2, .{});
 }
+
+pub fn generatePrime(global: *JSC.JSGlobalObject) callconv(.C) JSC.JSValue {
+    const S = struct {
+        fn cb(globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSC.JSValue {
+            const arguments = callframe.arguments(2).slice();
+
+            if (!arguments[0].isNumber()) return globalThis.throwInvalidArgumentTypeValue("bits", "unsigned integer", arguments[0]);
+            const bits_i64 = arguments[0].to(i64);
+
+            if (bits_i64 < 1 or bits_i64 > @as(i64, std.math.maxInt(i32))) {
+                globalThis.throwValue(globalThis.createInvalidArgs("bits must be a positive integer", .{}));
+                return .zero;
+            }
+
+            const bits: c_int = @as(c_int, @intCast(bits_i64));
+
+            var safe: bool = false;
+            var big_int: bool = false;
+
+            const add: [*c]BoringSSL.BIGNUM = null;
+            const rem: [*c]BoringSSL.BIGNUM = null;
+
+            const options_value = arguments[1];
+            if (!options_value.isEmptyOrUndefinedOrNull()) {
+                if (!options_value.isObject()) {
+                    globalThis.throwValue(globalThis.createInvalidArgs("options must be an object", .{}));
+                    return .zero;
+                }
+
+                if (options_value.get(globalThis, "safe")) |v| {
+                    if (!v.isBoolean()) {
+                        globalThis.throwValue(globalThis.createInvalidArgs("safe must be a boolean", .{}));
+                        return .zero;
+                    }
+                    safe = v.toBoolean();
+                }
+                if (options_value.get(globalThis, "bigint")) |v| {
+                    if (!v.isBoolean()) {
+                        globalThis.throwValue(globalThis.createInvalidArgs("bigint must be a boolean", .{}));
+                        return .zero;
+                    }
+                    big_int = v.toBoolean();
+                }
+                if (options_value.get(globalThis, "add")) |v| {
+                    // TODO: `add` read it in ect...
+                    _ = v;
+                }
+                if (options_value.get(globalThis, "rem")) |v| {
+                    // TODO: `rem` read it in ect...
+                    _ = v;
+                }
```

Comment:
these accept: `NodeJS.ArrayBufferView | SharedArrayBuffer | ArrayBuffer | bigint`

should we convert all to an `ArrayBuffer` in JS, then consume the arrayBuffer in zig?

Reply:
converted `bigint` -> `buffer in JS